sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1273,3 +1273,30 @@ def test_pairwise_distances_data_derived_params(n_jobs, metric, dist_function,
 
         assert_allclose(dist, expected_dist_explicit_params)
         assert_allclose(dist, expected_dist_default_params)
+
+
+@pytest.mark.parametrize(
+        "dtype",
+        [np.half, np.float, np.double, np.longdouble])
+@pytest.mark.parametrize("y_is_x", [True, False], ids=["Y is X", "Y is not X"])
+def test_pairwise_distances_input_datatypes(dtype, y_is_x):
```

Comment:
> This should probably mention seuclidean as it only tests that metric.

To clarify, you mean I should change the name of the test?  Otherwise it is mentioned in the following comments.

> Or, rather than adding a test, we should be modifying another test to ensure invariance to dtype across metrics

I had originally tried to extend this to include other metrics, but 'malahanobis' will fail on the Windows py35_pip_openblas_32bit benchmark because `np.linalg` doesn't like the usage of `np.double` presumably due to the 32bit openblas library.

Reply:
Then it might be best to test on all but xfail on some particularly problematic combinations.

At the moment it looks quite strange that you are only testing seuclidean here

Reply:
Please see my most recent revisions. I've added each of the non-boolean scipy distance metrics to the tests.  xfails have been introduced if each of the following hold:

- `metric == 'mahalanobis'`
- `dtype == np.longdouble`
- `sys.platform.startswith('win')`